src-tauri/src/farmer.rs

```diff
@@ -41,16 +39,16 @@ impl Default for ArchivingFrom {
 }
 
 #[tauri::command]
-pub(crate) async fn farming(path: String, reward_address: String, plot_size: u64) -> bool {
-    // create a channel to listen for farmer errors, and restart another farmer instance in case any error
-    let (error_sender, mut error_receiver): (Sender<()>, Receiver<()>) = channel(1);
-
+pub(crate) async fn farming(
+    path: String,
+    reward_address: String,
+    plot_size: u64,
+) -> Result<String, String> {
```

Comment:
@ozgunozerk why is it necessary to return anything other than `()` in success case?

Reply:
It is not necessary for sure. 
in the frontend, at some point we may want to log the success message from the backend, that was my thought. I can remove it

Reply:
@isSerge , your take on this? will it be helpful for frontend to provide a success message, or no?

Reply:
success message - not really, error maybe, but not critical at the moment